common/src/main/java/com/wynntils/models/players/GuildModel.java

```diff
@@ -117,13 +132,36 @@ public void parseGuildInfoFromGuildMenu(ItemStack guildInfoItem) {
             }
 
             Matcher rankMatcher = line.getMatcher(GUILD_RANK_MATCHER);
-
             if (rankMatcher.matches()) {
                 guildRank = GuildRank.valueOf(rankMatcher.group(1).toUpperCase(Locale.ROOT));
             }
         }
 
-        WynntilsMod.info("Successfully parsed guild info, " + guildRank + " of " + guildName);
+        WynntilsMod.info("Successfully parsed guild name and rank, " + guildRank + " of " + guildName);
+    }
+
+    public void parseGuildContainer(ContainerContent container) {
+        ItemStack membersItem = container.items().get(MEMBERS_SLOT);
+
+        Matcher levelMatcher = StyledText.fromComponent(membersItem.getHoverName())
+                .getNormalized()
+                .getMatcher(LEVEL_MATCHER);
+        if (!levelMatcher.matches()) {
+            WynntilsMod.warn("Could not parse guild level from item: " + LoreUtils.getLore(membersItem));
+            return;
+        }
+        guildLevel = Integer.parseInt(levelMatcher.group("level"));
+
+        Matcher levelProgressMatcher = LoreUtils.matchLoreLine(membersItem, 0, LEVEL_PROGRESS_MATCHER);
+        if (!levelMatcher.matches()) {
+            WynntilsMod.warn("Could not parse guild level progress from item: " + LoreUtils.getLore(membersItem));
+            return;
+        }
+        long current = Long.parseLong(levelProgressMatcher.group("current").replace(",", ""));
+        long required = Long.parseLong(levelProgressMatcher.group("required").replace(",", ""));
+        guildLevelProgress = new CappedValue((int) (((double) current / required) * 100d), 100);
+
+        WynntilsMod.info("Successfully parsed guild info for guild " + guildName);
```

Comment:
Parsing this screen has multiple benefits:
1. It sometimes takes a few seconds for the InfoBar to update on world change, so this is faster in 90% of the cases.
2. Having this in place allows for even more functions in the future (e. g. amount of held territories, boost level, members completed their objective, etc.) and more importantly to track tributes, which are currently not calculated in the territory management screen.

Reply:
I originally did not want to add more queries on class join, to reduce the time it needed for us to "init" the class, on very laggy servers/connections. I am still entirely convinced that we should add more queries. But I know that we can parse a lot from this container. So, the best thing we can do is optimize the amount of "clicks" needed by the container query.

Reply:
An alternative way of accessing this container would be "/guild manage", however I found that this was a bit unreliable because commands such as "/friend list" and "/use" would be run at the same time

Reply:
Running commands are no better, delay wise. Let's go with this for now.